lib/Mailer/Mailjet.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+namespace Stampie\Mailer;
+
+use Stampie\Adapter\ResponseInterface;
+use Stampie\Attachment;
+use Stampie\Exception\ApiException;
+use Stampie\Exception\HttpException;
+use Stampie\IdentityInterface;
+use Stampie\Mailer;
+use Stampie\Message\AttachmentsAwareInterface;
+use Stampie\Message\TaggableInterface;
+use Stampie\MessageInterface;
+use Stampie\Util\AttachmentUtils;
+
+/**
+ * @author Jérôme Parmentier <jerome@prmntr.me>
+ */
+class Mailjet extends Mailer
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function getEndpoint()
+    {
+        return 'https://api.mailjet.com/v3.1/send';
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function setServerToken($serverToken)
+    {
+        if (false === strpos($serverToken, ':')) {
+            throw new \InvalidArgumentException('Mailjet uses a "publicApiKey:privateApiKey" based ServerToken');
+        }
+
+        parent::setServerToken($serverToken);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getHeaders()
+    {
+        return [
+            'Accept' => 'application/json',
+            'Authorization' => sprintf('Basic %s', base64_encode($this->getServerToken())),
+            'Content-Type' => 'application/json',
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function format(MessageInterface $message)
+    {
+        $headers = [];
+        foreach ($message->getHeaders() as $name => $value) {
+            $headers[$name] = $value;
+        }
+
+        $parameters = [
+            'From' => $this->buildIdentityString($message->getFrom()),
+            'To' => $this->buildIdentityString($message->getTo()),
+            'Subject' => $message->getSubject(),
+            'Headers' => $headers,
+            'HTMLPart' => $message->getHtml(),
+            'TextPart' => $message->getText(),
+            'ReplyTo' => $this->buildIdentityString($message->getReplyTo()),
+        ];
+
+        $attachments = $this->getFiles($message);
+
+        if ($attachments['attached']) {
+            $parameters['Attachments'] = $attachments['attached'];
+        }
+
+        if ($attachments['inlined']) {
+            $parameters['InlinedAttachments'] = $attachments['inlined'];
+        }
+
+        if ($message instanceof TaggableInterface) {
+            $tag = $message->getTag();
+
+            if (is_array($tag)) {
+                $tag = reset($tag);
+            }
+
+            $parameters['EventPayload'] = $tag;
```

Comment:
the documentation of EventPayload on mailjet looks like it better fits for the Metadata concept than the Tag one btw

Reply:
Tags would fit better in `MonitoringCategory` IMO (supporting a single one indeed)